### PR TITLE
Individual forums replies network search issue - fixed

### LINKED ISSUE #2453: Network search> Individual forums replies related search data is not appearing under the search suggestions and on search result page.
**Describe the bug**
If in network search > settings> Forum replies option is enabled >
Then search data is not appearing as per Public individual forum replies. Data is not appearing in suggestion as well as on the search result page where as if we search replies of Public Group discussion then it is appearing in suggestion as well as on the search result page.

**To Reproduce**
1. Go to 'Settings> Search> Check Forums and Replies.'
2. Search using Public  individual forum discussion reply. 
3. See error.

**Expected behavior**
Search result must appear for Replies of Public individual forums.

**Screenshots**
https://www.loom.com/share/ae7d47832c844a6ab7eea810988a0721

**Support ticket links**
N/A internally reported by QA

### PR DESCRIPTION
**Pull request by mahedihasannoman**
### All Submissions:

* [x] Have you followed the [Pull Requests Guidelines](https://github.com/buddyboss/buddyboss-platform/wiki/Submitting-Pull-Requests#pull-request-guidelines)?
* [x] Does your code follow the [WordPress' Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Fixes #2453  .

### How to test the changes in this Pull Request:

1. Go to 'Settings> Search> Check Forums and Replies.'
2. Search using Public individual forum discussion reply.
3. See it works.

### Proof Screenshots or Video

https://www.loom.com/share/1fb80ee481284c22963b842411ff6718

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Enter a summary of all changes on this Pull Request. This will appear in the changelog if accepted.

### CODE REVIEW

File: src/bp-search/classes/class-bp-search-bbpress-forums-replies.php [hunk 1]
```diff
@@ -83,15 +91,7 @@ function ( $group_id ) {
...
-				$group_query = ' pm.meta_value IN ( SELECT post_id FROM ' . $wpdb->postmeta . ' WHERE meta_key = \'_bbp_group_ids\' AND meta_value IN(' . trim( $in, ',' ) . ') ) AND ';
-			}
-
-			if ( current_user_can( 'read_hidden_forums' ) ) {
-				$post_status = array( "'publish'", "'private'", "'hidden'" );
-			} elseif ( current_user_can( 'read_private_forums' ) ) {
-				$post_status = array( "'publish'", "'private'" );
-			} else {
-				$post_status = array( "'publish'" );
+				$group_query = ' ( pm.meta_value IN ( SELECT post_id FROM ' . $wpdb->postmeta . ' WHERE meta_key = \'_bbp_group_ids\' AND meta_value IN(' . trim( $in, ',' ) . ') ) OR pm.meta_value IN( SELECT post_id FROM ' . $wpdb->postmeta . ' WHERE meta_key != \'_bbp_group_ids\' and post_id in ( SELECT ID FROM ' . $wpdb->posts . ' WHERE post_type = \'forum\' AND post_status IN (' . join( ',', $post_status ) . ') ) ) ) AND ';
```
Comment (dipskakadiya) [inline src/bp-search/classes/class-bp-search-bbpress-forums-replies.php:94]: @mahedihasannoman Why we need to inner Query 
I think simplest solution is just add `post status` with query

**Review by dipskakadiya — CHANGES_REQUESTED** [review]

File: src/bp-search/classes/class-bp-search-bbpress-forums-replies.php (hunk 1 above)
Comment (mahedihasannoman) [reply src/bp-search/classes/class-bp-search-bbpress-forums-replies.php]: Hi @dipskakadiya, Done as we discussed.

**Review by dipskakadiya — APPROVED** [review]

**Review by KartikSuthar — APPROVED** [review]

**Comment by sadrul** [comment]
@sumitbuddyboss @mohit-buddyboss merged into dev and qa-ongoing and ready for testing

**Comment by JetalDobariya** [comment]
Hello @sadrul 

I did verify this PR in qa-ongoing branch , now individual forum search results are getting proper.

See in the Video : https://www.loom.com/share/760c0a87a86f4206be4543231d593b7b

Thanks!

**Comment by JetalDobariya** [comment]
Hello @sadrul 

I did verify this PR in qa-approved branch , now individual Forum search results are getting proper.
--> https://www.loom.com/i/765e9b38856747e0a9407e66ca63c8ea

Thanks!

**Pull request merged**